### DISCUSSION TITLE
Binary sensor bar chart "on" duration

### DISCUSSION THREAD

**Discussion by Chavife** [dbuezas/lovelace-plotly-graph-card#286] [Show and tell]
I want to create a bar chart from my binary sensor which would show the duration of `on` time in bar charts.
The use case is to track the "Do not disturb" sensor from my phone which I turn on/off when going to sleep and I wake up.
I want to see how much I slept each day.

The sensor is changing from `off` to `priority_only`

![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/18663896/f075653d-a87c-4e1d-9fde-ea2270042670)

I cant figure out how to show the durations of the `priority_only` durations in bar chart.

**Comment by dbuezas** [comment]
Something like this;
```yaml
type: custom:plotly-graph
hours_to_show: 10d
entities:
  - entity: switch.teich_pumpe_switch
    unit_of_measurement: hours
    filters:
      - resample: 1m
      - fn: |
          ({xs, ys}) => {

            const result = {
              xs: [],
              ys: [],
            };
            let last = new Date(xs[0]);
            last.setHours(0,0,0,0);
            let hours = 0
            for (let i = 0; i < xs.length+1; i++){
              let curr = new Date(xs[i]);
              curr.setHours(0,0,0,0);
              if (+last !== +curr) {
                result.xs.push(last);
                result.ys.push(hours);
                last = curr;
                hours = 0;
              }
              if (ys[i] === "on") hours += 1/60;
            } 
            return result;
          }
```

It resamples to every minute so we can count instead of subtracting timestamps (lazy but it works), then just count how many datapoints are "on", and add them all up by day.

**Reply by Chavife** [reply]
works like a charm, really nice idea. Thanks for the help

**Reply by dbuezas** [reply]
I just realized you may want to use a time_offset of 12h. This way you get the night and next morning accounted for as 1 day
